### DISCUSSION TITLE
How to make clients connect to pgcat with 'sslmode=require'?

### DISCUSSION THREAD

**Discussion by civilizeddev** [postgresml/pgcat#703] [Q&A]
If you read the [README.md](https://github.com/postgresml/pgcat/blob/358724f7a979ce5e5f0cb211873125865dccf220/README.md?plain=1#L21), it states that a secure connection from the client to **pgcat** is possible.

```
Clients can connect to the pooler using TLS.
```

### sslmode=require

![](https://www.plantuml.com/plantuml/svg/DSx13S8m38NXkwTu2tG1951N2SaZH4hZOhjS4BjJgBX--cszsIDPR4iNwRm5Gc6jtWEb6NCq7I8fXnfoRnmXykKqQYUSzgJX9K3MPTN14LXhJh5BIP6koObLFQhHvIro_C0qNOuE9xZtHGlFniVMZ7Buh_WVoinHzeaV)

So, I tried to connect to **pgcat** from the client with the `sslmode=require` option.

```
"host=localhost port=5432 database=pgcat user=admin_user password=admin_pass sslmode=require"
```

However, it seems that **pgcat** does not accept `sslmode=require`.

```
panic: pg: SSL is not enabled on the server
```

<img width="642" alt="Screenshot 2024-03-07 at 12 33 16" src="https://github.com/postgresml/pgcat/assets/15084045/d08c061e-1061-4b9d-b460-2fc5d99e4899">

It appears that just the [server_tls](https://github.com/postgresml/pgcat/blob/main/CONFIG.md#server_tls
) option in **pgcat.toml** is not enough to enable **pgcat** to support `sslmode=require`.

As you can see, the log says `"TLS support is disabled"` when **pgcat** is run.

<img width="522" alt="Screenshot 2024-03-07 at 12 30 31" src="https://github.com/postgresml/pgcat/assets/15084045/f63bcc58-a738-4770-a58d-45290cdcca9f">

How can I make it possible for the client to connect to **pgcat** with `sslmode=require`?

I would appreciate it if you could tell me how.

**Comment by wata727** [comment]
You may be able to enable TLS connection between client and PgCat by setting `tls_certificate` and `tls_private_key`.
https://github.com/postgresml/pgcat/blob/v1.2.0/src/config.rs#L325-L326
